packages/client/components/StandardHubUserMenu.tsx

```diff
@@ -30,92 +30,90 @@ const TallMenu = styled(Menu)({
   maxHeight: 256
 })
 
-interface Props extends RouteComponentProps<{[x: string]: string | undefined}> {
+const MenuItemLink = MenuItemLabel.withComponent(Link)
+
+interface Props {
   menuProps: MenuProps
-  viewer: StandardHubUserMenu_viewer | null
+  viewerRef: StandardHubUserMenu_viewer$key
 }
 
 const StandardHubUserMenu = (props: Props) => {
-  const {menuProps, history, viewer} = props
-  if (!viewer) return null
-  const {email, organizations} = viewer
-  // nav menu routes
-  const goToProfile = () => {
-    history.push('/me/profile')
-  }
-  const goToOrganizations = () => {
-    history.push('/me/organizations')
-  }
-  const signOut = () => {
-    history.push(`/${SIGNOUT_SLUG}`)
-  }
-
+  const {menuProps, viewerRef} = props
+  const viewer = useFragment(
+    graphql`
+      fragment StandardHubUserMenu_viewer on User {
+        email
+        featureFlags {
+          insights
+        }
+        organizations {
+          id
+          tier
+        }
+      }
+    `,
+    viewerRef
+  )
+  const {email, featureFlags, organizations} = viewer
+  const {insights} = featureFlags
   const ownedFreeOrgs = organizations.filter((org) => org.tier === 'personal')
   const showUpgradeCTA = ownedFreeOrgs.length > 0
-
-  const handleUpgradeClick = () => {
-    const routeSuffix = ownedFreeOrgs.length === 1 ? `/${ownedFreeOrgs[0]!.id}` : ''
-    history.push(`/me/organizations${routeSuffix}`)
-  }
+  const routeSuffix = ownedFreeOrgs.length === 1 ? `/${ownedFreeOrgs[0]!.id}` : ''
 
   return (
     <TallMenu ariaLabel={'Select your settings'} {...menuProps}>
       <DropdownMenuLabel>{email}</DropdownMenuLabel>
       <MenuItem
         label={
-          <MenuItemLabel>
+          <MenuItemLink to={'/me/profile'}>
             <MenuItemIcon>account_box</MenuItemIcon>
             {'Profile'}
-          </MenuItemLabel>
+          </MenuItemLink>
         }
-        onClick={goToProfile}
       />
       <MenuItem
         label={
-          <MenuItemLabel>
+          <MenuItemLink to={'/me/organizations'}>
             <MenuItemIcon>account_balance</MenuItemIcon>
             {'Organizations'}
-          </MenuItemLabel>
+          </MenuItemLink>
         }
-        onClick={goToOrganizations}
       />
+      {insights && (
```

Comment:
+1 should we check the length of `domains` too? If the feature flag is turned on but there are no active domains, users will see an empty page